pkg/pillar/types/cipherinfotypes.go

```diff
@@ -10,54 +10,81 @@ import (
 	zcommon "github.com/lf-edge/eve/api/go/evecommon"
 )
 
-// CipherContext : a pair of device and controller certificate
-// published by controller along with some attributes
-// part of EdgeDevConfig block, received from controller
-type CipherContext struct {
+// EncryptionBlock - This is a Mirror of
+// api/proto/config/acipherinfo.proto - EncryptionBlock
+// Always need to keep these two consistent.
+type EncryptionBlock struct {
+	DsAPIKey          string
+	DsPassword        string
+	WifiUserName      string // If the authentication type is EAP
+	WifiPassword      string
+	ProtectedUserData string
+}
+
+// CipherContextConfig : a pair of device and controller certificate
+// published by controller along with some attributes // part of EdgeDevConfig block, received from controller
+type CipherContextConfig struct {
 	ContextID          string
 	HashScheme         zcommon.HashAlgorithm
 	KeyExchangeScheme  zconfig.KeyExchangeScheme
 	EncryptionScheme   zconfig.EncryptionScheme
 	ControllerCertHash []byte
 	DeviceCertHash     []byte
-	// ErrorAndTime provides SetErrorNow() and ClearError()
+}
+
+// Key :
+func (config *CipherContextConfig) Key() string {
+	return config.ContextID
+}
+
+// ControllerCertKey :
+func (config *CipherContextConfig) ControllerCertKey() string {
+	return hex.EncodeToString(config.ControllerCertHash)
+}
+
+// EveNodeCertKey :
+func (config *CipherContextConfig) EveNodeCertKey() string {
+	return hex.EncodeToString(config.DeviceCertHash)
+}
+
+// CipherContextStatus : context information for the pair
+// of certificates
+type CipherContextStatus struct {
+	CipherContextConfig
+	ControllerCert []byte // resolved through CERT API
 	ErrorAndTime
```

Comment:
This is not needed since we have a working existing mechanism to pass the controller cert(s). This PR should not change that. It has nothing to do with the controller certs, but with creating, publishing and using the edge node cert for ECDH in such a way it can easily be extended to publish the attestation cert. And then using the ECDH cert for decrypt. I don't expect that to me more than 200 lines of diffs total.